src/message_ref.rs

```diff
@@ -210,21 +214,38 @@ impl<'a> RequestRef<'a> {
 }
 
 #[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde_serialize", derive(serde::Serialize))]
 pub(crate) struct RequestLine<'a> {
     pub(crate) method: MethodRef<'a>,
     pub(crate) request_uri: Option<&'a str>,
     pub(crate) version: Version,
 }
 
 #[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde_serialize", derive(serde::Serialize))]
 pub struct ResponseRef<'a> {
     pub(crate) version: Version,
     pub(crate) status: StatusCode,
     pub(crate) reason_phrase: &'a str,
+    #[cfg_attr(feature = "serde_serialize", serde(serialize_with = "serialize_tinyvec"))]
     pub(crate) headers: TinyVec<[HeaderRef<'a>; 16]>,
     pub(crate) body: &'a [u8],
 }
 
+#[cfg(feature = "serde_serialize")]
+fn serialize_tinyvec<S, T>(tinyvec: &TinyVec<[T; 16]>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
If you enable the `serde` feature on `tinyvec` then this shouldn't be necessary

Reply:
I will make this change, I didn't bother checking.  I will make the changes.